pandas/core/reshape/pivot.py

```diff
@@ -507,17 +511,30 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
-                               aggfunc=len, margins=margins,
-                               margins_name=margins_name, dropna=dropna)
+                               aggfunc=len, margins=False, dropna=dropna)
         table = table.fillna(0).astype(np.int64)
 
     else:
         data['__dummy__'] = values
         df = DataFrame(data)
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
-                               aggfunc=aggfunc, margins=margins,
-                               margins_name=margins_name, dropna=dropna)
+                               aggfunc=aggfunc, margins=False, dropna=dropna)
 
+    if margins:
+        _check_margins_name(margins_name, table)
+
+        if normalize != 'index':
+            # add margin column
+            table[margins_name] = table.sum(axis=1)
+
+        if normalize != 'columns':
+            # add margin row
+            if type(table.index) is MultiIndex:
+                table = table.transpose()
+                table[margins_name] = table.sum(axis=1)
```

Comment:
This solution does not work: it flattens the MultiIndex.
Strangely, it works for columns. Therefore I did this workaround using transpose().
Do you have another idea how to deal with that?
    
    import pandas as pd
    import numpy as np
    a = np.array(['foo', 'bar', 'foo', 'bar', 'bar', 'foo'])
    b = np.array(['one', 'one', 'two', 'one', 'two', 'two'])
    c = np.array(['dull', 'shiny', 'dull', 'dull', 'dull', 'shiny'])
    d = np.array(['a', 'a', 'b', 'a', 'b', 'b'])

    #dataframe with mutliindex columns and multiindex index 
    df = pd.crosstab([a, b], [c, d], normalize='columns',
                             margins=False)

    #this works
    df['all'] = df.sum(axis=1)

    #this destroys the multiindex
    df.loc['all'] = df.sum(axis=0)


    print(df)
    col_0      dull      shiny       all
    col_1         a    b     a    b     
    (bar, one)  0.5  0.0   1.0  0.0  1.5
    (bar, two)  0.0  0.5   0.0  0.0  0.5
    (foo, one)  0.5  0.0   0.0  0.0  0.5
    (foo, two)  0.0  0.5   0.0  1.0  1.5
    all         1.0  1.0   1.0  1.0  4.0